templates/flavors/base/cluster-template.yaml

```diff
@@ -29,6 +29,19 @@ spec:
   networkSpec:
     vnet:
       name: "${AZURE_VNET_NAME:=${CLUSTER_NAME}-vnet}"
+      cidrBlocks:
+        - 10.0.0.0/8
+    subnets:
+    - cidrBlocks:
+      - 10.0.0.0/16
+      name: control-plane-subnet
+      role: control-plane
+    - cidrBlocks:
+      - 10.1.0.0/16
```

Comment:
is the idea to make vnet / subnet cidr ranges more explicit in templates? These values aren't actually required to be set as far as I understand

Reply:
I just copied the relevant configuration stuffs from the existing NAT Gateway template spec.

Happy to make it more concise if that's better.

Reply:
I don't think it's necessary to actually include it in here. It will get defaulted to these same values if not specified. I guess the upside on including it is that it makes the configuration more explicit and transparent, the downside is that it's less concise. WDYT?

Reply:
@jackfrancis PTAL when you get a chance

Reply:
Is this a correct, more concise definition to put into the base and then apply everywhere?:

```
    vnet:
      name: "${AZURE_VNET_NAME:=${CLUSTER_NAME}-vnet}"
    subnets:
    - name: control-plane-subnet
      role: control-plane
    - name: node-subnet
      natGateway:
        name: node-natgateway
      role: node
```

Reply:
yes that looks good